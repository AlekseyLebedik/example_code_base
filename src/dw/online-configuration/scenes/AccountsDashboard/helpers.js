import { useMemo } from 'react';
import moment from 'moment-timezone';
import get from 'lodash/get';
import isEmpty from 'lodash/isEmpty';
import flatMap from 'lodash/flatMap';
import groupBy from 'lodash/groupBy';
import orderBy from 'lodash/orderBy';
import maxBy from 'lodash/maxBy';

import deepCopy from 'dw/core/helpers/deep-copy';

import { COUNTRY_CODE_IDX, DISPLAY_NAMES_MAP, VALUE_IDX } from './constants';

import styles from './index.module.css';

/**
 * Get latest date in KPI Data
 * @param {*} kpiList
 */
export const getLatestUpToDate = kpiList =>
  flatMap(kpiList.map(d => d.details.data)).sort(
    (item1, item2) =>
      moment(item2[0], 'YYYY-MM-DD').unix() -
      moment(item1[0], 'YYYY-MM-DD').unix()
  )[0][0];

/**
 * Get message for displaying how up to date KPIs are (last updated)
 * @param {*} kpiList
 */
export const getLatestUpToDateMsg = kpiList =>
  kpiList.length > 0 ? `Last updated ${getLatestUpToDate(kpiList)}` : null;

/**
 * Transform details for KeyValue table
 * @param {*} kpis
 * @param {*} kpiDetails
 */
export const useTransformDetails = (kpis, kpiDetails) => {
  const displayName = useMemo(() => {
    if (DISPLAY_NAMES_MAP.hasOwnProperty(kpiDetails?.name))
      return DISPLAY_NAMES_MAP[kpiDetails?.name];
    const kpi = kpis && kpis.find(({ name }) => name === kpiDetails?.name);
    // eslint-disable-next-line
    return kpi?.display_name || kpiDetails?.name;
  }, [kpis, kpiDetails]);
  const hasTwoDimensions =
    kpiDetails?.details?.fieldNames?.includes('second_dimension');
  const data = useMemo(() => {
    const grouped = hasTwoDimensions
      ? groupBy(kpiDetails?.details?.data, row => row[1])
      : { total: kpiDetails?.details?.data };
    const normalizedData = Object.entries(grouped).reduce(
      (acc, [key, items]) => {
        const maxItem = maxBy(items, ([date]) =>
          moment(date, 'YYYY-MM-DD').unix()
        );
        return [...acc, [key, maxItem ? maxItem[maxItem.length - 1] : 0]];
      },
      []
    );
    return normalizedData;
  }, [kpiDetails, hasTwoDimensions]);
  const total = useMemo(
    () => data.reduce((acc, row) => acc + row[row.length - 1], 0),
    [data]
  );
  return {
    displayName,
    data,
    name: kpiDetails?.name,
    hasTwoDimensions,
    total,
  };
};

/**
 * Get default for kpi details
 * @param {*} getDetails
 * @param {*} name
 */
export const getDetailsDefault = (getDetails, name) =>
  getDetails(name) || {
    name,
    details: { data: [], fieldNames: [] },
  };

/**
 * Merge kpis
 * @param {*} notOk
 * @param {*} ok
 * @param {*} underAge
 */
export const mergeKpis = (notOk, ok, underAge = null, name) => {
  const result = deepCopy(notOk);
  result.name = name;
  result.displayName = DISPLAY_NAMES_MAP[name] || name;
  if (result.hasTwoDimensions) {
    result.data.forEach(row => {
      const [key] = row;
      const other = ok.data.find(([otherKey]) => otherKey === key);
      const value = other ? other[other.length - 1] : 0;
      row.push(value);
    });
  } else {
    // eslint-disable-next-line
    result.data = [['Not ok to sell', notOk.total]];
    if (underAge) {
      result.data.push(['Under 16', underAge.total]);
    }
    result.data.push(['Ok to sell', ok.total]);
  }
  return result;
};

/**
 * Sort countries
 * @param {*} row
 */
export const sortCountries = row => {
  const countryCode = row[COUNTRY_CODE_IDX];
  if (countryCode === 'US') return 10 ** 9 + 1;
  if (countryCode === 'GB') return 10 ** 9;
  return row[VALUE_IDX];
};

/**
 * Get country name for country code
 * @param {*} row
 */
export const getCountryName = row => {
  const regionNamesInEnglish = new Intl.DisplayNames(['en'], {
    type: 'region',
  });
  const newRow = [...row];
  try {
    newRow[COUNTRY_CODE_IDX] = regionNamesInEnglish.of(row[COUNTRY_CODE_IDX]);
  } catch (e) {
    // eslint-disable-next-line
    console.log(`Cannot find name for country code ${row[COUNTRY_CODE_IDX]}`);
  }
  return newRow;
};

/**
 * Replace country codes with country names for data
 * @param {*} kpiDetails
 */
export const useCountryNames = kpiDetails => {
  if (!kpiDetails?.name?.includes('per_country')) {
    return kpiDetails;
  }
  const details = {
    ...kpiDetails,
    data: orderBy(kpiDetails.data, sortCountries, 'desc')
      .filter(item => !!item[COUNTRY_CODE_IDX])
      .map(getCountryName),
  };
  return details;
};

/**
 * Get display value for per country age group
 * @param {*} getDetails
 * @param {*} name
 * @param {*} selectedCountry
 */
export const useAgeGroup = (getDetails, name, selectedCountry) => {
  const details = getDetailsDefault(getDetails, name);
  const normalized = useTransformDetails(null, details);
  const foundCountry = normalized.data.find(
    item => getCountryName(item) && getCountryName(item)[0] === selectedCountry
  );
  const value = useMemo(() => {
    if (selectedCountry) {
      if (foundCountry) return foundCountry[1];
      return 0;
    }
    return normalized.data.reduce((acc, row) => acc + row[row.length - 1], 0);
  }, [normalized, selectedCountry, foundCountry]);

  return value;
};

/**
 * Transform privacy KPI data into format for KeyValue component
 * @param {*} data
 */
export const transformPrivKpiData = data => {
  if (isEmpty(data)) return {};
  const {
    query_results: { column_positions: cols, row_positions: rows, values },
  } = data;
  const transformedData = {};
  cols.forEach((col, colIdx) => {
    const kpiTypeData = [];
    values.forEach((valueList, valIdx) => {
      kpiTypeData.push([rows[valIdx][0].name, valueList[colIdx]]);
    });
    transformedData[col[col.length - 1].name] = kpiTypeData;
  });
  return transformedData;
};

/**
 * Filter Privacy KPI Data, sort by date, then format for display
 * @param {*} data
 */
const filterSortPrivKpiData = data =>
  data
    .filter(row => row[0] !== '$total')
    .sort(
      (item1, item2) =>
        moment(item2[0], 'MMM YYYY').unix() -
        moment(item1[0], 'MMM YYYY').unix()
    )
    .map(item => [moment(item[0], 'MMM YYYY').format('MMMM'), item[1]]);

/**
 * Transform Privacy KPI data for display
 * according to the kpi name, assign rest of
 * KeyValue props according to the particular
 * kpi name
 * @param {*} kpiName
 */
export const formatPrivKpiData = (kpiName, privacyKpiData, totals) => {
  switch (kpiName) {
    case 'privacy_kpi_delete_requests': {
      const rtbfData = privacyKpiData['RTBF Sub-Task'];
      return {
        data: filterSortPrivKpiData(rtbfData),
        header: [
          'Total # of Deletes',
          get(totals, ['RTBF Sub-Task', '0', '1'], ''),
        ],
        itemsToDisplay: 2,
        keySize: 8,
        overrideTableStyle: styles.rtbfContainer,
        sortDataByDate: true,
      };
    }
    case 'privacy_kpi_export_requests': {
      const dprData = privacyKpiData['DPR Sub-Task'];
      const sarData = privacyKpiData['SAR Sub-Task'];
      const exportData = [...dprData, ...sarData].reduce((acc, row) => {
        const foundIdx = acc.findIndex(item => item[0] === row[0]);
        if (foundIdx > -1) {
          const updatedAcc = [...acc];
          updatedAcc[foundIdx] = [
            updatedAcc[foundIdx][0],
            updatedAcc[foundIdx][1] + row[1],
          ];
          return updatedAcc;
        }
        return [...acc, row];
      }, []);
      const exportTotals =
        get(totals, ['DPR Sub-Task', '0', '1'], '') +
        get(totals, ['SAR Sub-Task', '0', '1'], '');
      return {
        data: filterSortPrivKpiData(exportData),
        header: ['Total # of Exports', exportTotals],
        itemsToDisplay: 2,
        keySize: 8,
        overrideTableStyle: styles.dprSarContainer,
        sortDataByDate: true,
      };
    }
    default: {
      return {
        data: [],
        header: [
          'Total # of Privacy requests to date',
          get(totals, ['$total', '0', '1'], ''),
        ],
        itemsToDisplay: 0,
        keySize: 10,
        overrideTableStyle: styles.totalsContainer,
        viewAllButtonName: 'TOTAL PER COUNTRY',
      };
    }
  }
};

/**
 * Determines name of view all button. When showDataModalOnly
 * (when data shows only in view modal), use the header name
 * as the view all button name, otherwise just display 'VIEW ALL'.
 * Adding viewAllButtonName overrides these names.
 * @param {*} showDataModalOnly
 * @param {*} keyHeader
 * @param {*} viewAllButtonName
 */
export const getViewAllButtonName = (
  showDataModalOnly,
  keyHeader,
  viewAllButtonName = null
) => {
  if (viewAllButtonName) return viewAllButtonName;
  return showDataModalOnly ? `${keyHeader}` : 'VIEW ALL';
};
