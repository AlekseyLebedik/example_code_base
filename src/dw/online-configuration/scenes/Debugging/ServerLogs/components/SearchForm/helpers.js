import { LOG_LEVELS, SOURCE_TYPES } from './constants';

export const parseBooleanValues = query =>
  Object.keys(query).reduce((acc, item) => {
    if (LOG_LEVELS.includes(item) || SOURCE_TYPES.includes(item)) {
      return {
        ...acc,
        [item]: query[item] === 'true',
      };
    }
    return acc;
  }, {});
