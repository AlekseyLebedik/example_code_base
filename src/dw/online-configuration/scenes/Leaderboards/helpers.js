/**
 * Search through retrieved leaderboard data and ensure that selected leaderboard
 * is in fetched data
 * @param {*} leaderboards
 * @param {*} selectedLeaderboardId
 * @returns
 */
export const selectedLeaderboardDataFetched = (
  leaderboards,
  selectedLeaderboardId
) =>
  Boolean(
    !!selectedLeaderboardId &&
      leaderboards.find(lb => lb.id === selectedLeaderboardId)
  );
