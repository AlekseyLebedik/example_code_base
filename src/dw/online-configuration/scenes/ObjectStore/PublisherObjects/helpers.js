import { BYTE_TO_GB, MAX_ZIP_DOWNLOAD_GB_SIZE } from './constants';

export function normalizeObjects(objects) {
  return objects.reduce(
    (map, metadata) => ({ ...map, [metadata.name]: metadata }),
    {}
  );
}

export function filterObject(obj, callback) {
  return Object.entries(obj).reduce(
    (map, [key, value]) => ({
      ...map,
      [key]: callback(value) ? value : undefined,
    }),
    {}
  );
}

export const getFileData = action => {
  const { formData } = action;
  const { size } = formData.get('fileData');
  return { size, fileName: formData.get('fileName') };
};

export const canDownloadFileGroupSize = files => {
  const contentLength = files.reduce(
    (acc, f) => acc + parseInt(f.contentLength, 10),
    0
  );
  return contentLength < MAX_ZIP_DOWNLOAD_GB_SIZE * BYTE_TO_GB;
};
