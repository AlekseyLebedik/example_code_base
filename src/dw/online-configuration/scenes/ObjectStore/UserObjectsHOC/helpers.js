export function normalizeObjects(objects) {
  return objects.reduce((map, data) => {
    const fileName = data.name || data.metadata.name;
    return {
      ...map,
      [fileName]: data,
    };
  }, {});
}

export function filterObject(obj, callback) {
  return Object.entries(obj).reduce(
    (map, [key, value]) => ({
      ...map,
      [key]: callback(value) ? value : undefined,
    }),
    {}
  );
}

export function sortDeletedObjects(results) {
  const { deletedObjects, notDeletedObjects } = Object.entries(results).reduce(
    (output, [name, deleted]) => {
      if (deleted) output.deletedObjects.push(name);
      else output.notDeletedObjects.push(name);
      return output;
    },
    { deletedObjects: [], notDeletedObjects: [] }
  );
  return { deletedObjects, notDeletedObjects };
}
