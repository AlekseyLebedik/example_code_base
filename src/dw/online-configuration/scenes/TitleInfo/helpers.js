export const currentEnv = state =>
  state.Components.TitleSelector.currentEnv.shortType;

export const currentTitle = state =>
  state.Components.TitleSelector.currentTitle.id;

export const titlePath = state =>
  `titles/${currentTitle(state)}/envs/${currentEnv(state)}/`;

export const createApiUrl = state => titlePath(state);
