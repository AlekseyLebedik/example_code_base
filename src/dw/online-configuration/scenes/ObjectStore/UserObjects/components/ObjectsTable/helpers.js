import { formatFileSize } from 'dw/core/helpers/formatters';
import { comparatorFilter } from 'dw/core/helpers/aggrid';
import {
  categoryRenderer,
  downloadRenderer,
  extraRenderer,
} from '../../../cellRenderers';

const components = {
  downloadRenderer,
  categoryRenderer,
  extraRenderer,
};

const valueAfterGroup = (field, valA, valB, nodeA, nodeB) => {
  const valueA =
    valA ||
    (nodeA?.childrenAfterGroup ? nodeA?.childrenAfterGroup[0].data[field] : 0);
  const valueB =
    valB ||
    (nodeB?.childrenAfterGroup ? nodeB?.childrenAfterGroup[0].data[field] : 0);
  return {
    valueA,
    valueB,
  };
};

const childrenAfterGroupComparator = field => (valA, valB, nodeA, nodeB) => {
  const { valueA, valueB } = valueAfterGroup(field, valA, valB, nodeA, nodeB);

  return valueA - valueB;
};

const getColumnDefinitions = (formatDateTime, isMaster) =>
  [
    isMaster && {
      headerName: 'Name',
      width: 500,
      cellRenderer: 'agGroupCellRenderer',
      field: 'name',
      checkboxSelection: true,
      headerCheckboxSelection: true,
    },
    isMaster && {
      headerName: 'Created',
      field: 'created',
      valueFormatter: params =>
        !params.value ? '' : formatDateTime(params.value).toString(),
      cellStyle: {
        'white-space': 'normal',
        'word-wrap': 'break-word',
      },
      width: 250,
      sortable: true,
      sort: 'desc',
      filter: 'agTextColumnFilter',
      filterParams: {
        filterOptions: ['contains'],
        textMatcher: (filter, value, filterText) =>
          comparatorFilter(
            filter,
            formatDateTime(value).toString(),
            filterText
          ),
      },
      comparator: childrenAfterGroupComparator('created'),
      getQuickFilterText: params => formatDateTime(params.value).toString(),
    },
    {
      headerName: 'Modified',
      field: 'modified',
      valueFormatter: params =>
        !params.value ? '' : formatDateTime(params.value).toString(),
      filter: 'agTextColumnFilter',
      filterParams: {
        filterOptions: ['contains'],
        textMatcher: (filter, value, filterText) =>
          comparatorFilter(
            filter,
            formatDateTime(value).toString(),
            filterText
          ),
      },
      comparator: childrenAfterGroupComparator('modified'),
      cellStyle: {
        'white-space': 'normal',
        'word-wrap': 'break-word',
      },
      sortable: true,
      width: isMaster ? 250 : 350,
      getQuickFilterText: params => formatDateTime(params.value).toString(),
    },
    isMaster && {
      headerName: 'Expires On',
      field: 'expiresOn',
      width: 150,
      valueFormatter: params =>
        !params.value ? '' : formatDateTime(params.value).toString(),
      filter: 'agTextColumnFilter',
      filterParams: {
        filterOptions: ['contains'],
        textMatcher: (filter, value, filterText) =>
          comparatorFilter(
            filter,
            formatDateTime(value).toString(),
            filterText
          ),
      },
      cellStyle: {
        'white-space': 'normal',
        'word-wrap': 'break-word',
      },
    },
    {
      headerName: 'Tags',
      field: 'tags',
      width: 450,
      sortable: true,
      sort: 'desc',
      filter: 'agTextColumnFilter',
      filterParams: {
        filterOptions: ['contains'],
      },
      valueGetter: params =>
        !params.data?.tags
          ? ''
          : params.data.tags
              .map(
                ({ key, value, values }) =>
                  `${key}:${(values && values.join(', ')) || value}`
              )
              .join(', '),
    },
    {
      headerName: 'Checksum',
      field: 'checksum',
      width: 450,
      cellStyle: {
        'white-space': 'normal',
        'word-wrap': 'break-word',
      },
    },
    isMaster && {
      headerName: 'Category',
      field: 'category',
      width: 250,
      filter: true,
      cellStyle: {
        'white-space': 'normal',
        'word-wrap': 'break-word',
      },

      valueFormatter: params => {
        if (params.value === null) {
          return 'N/A';
        }
        if (params.value) {
          return params.value;
        }
        return '';
      },
    },
    {
      headerName: 'Content Length',
      field: 'contentLength',
      width: 200,
      sortable: true,
      cellStyle: {
        'white-space': 'normal',
        'word-wrap': 'break-word',
      },
      valueFormatter: params =>
        !params.value ? '' : formatFileSize(params.value),
      comparator: childrenAfterGroupComparator('contentLength'),
    },
    {
      headerName: 'ACL',
      field: 'acl',
      width: 150,
      cellStyle: {
        'white-space': 'normal',
        'word-wrap': 'break-word',
      },
    },
    {
      headerName: 'extraData',
      field: 'extraData',
      cellRenderer: 'extraRenderer',
      width: 300,
      cellStyle: {
        'white-space': 'normal',
        'word-wrap': 'break-word',
      },
      valueFormatter: params => params.value || 'N/A',
    },
    {
      headerName: 'Actions',
      field: 'ddl',
      width: isMaster ? 200 : 180,
      minWidth: 200,
      cellRenderer: 'downloadRenderer',
      pinned: 'right',
    },
  ]
    .filter(item => item)
    .map(item => ({
      ...item,
      cellStyle: isMaster ? {} : { 'background-color': '#eeebeb' },
    }));

const getMasterDetailsConf = (formatDateTime, context) => ({
  masterDetail: true,
  detailCellRendererParams: {
    detailGridOptions: {
      columnDefs: getColumnDefinitions(formatDateTime),
      components,
      context,
      onFirstDataRendered: params => params.api.sizeColumnsToFit(),
      localeText: {
        noRowsToShow: 'There are no Backups yet for this User Object.',
      },
    },
    getDetailRowData(params) {
      params.successCallback(
        params.data.backups.map(item => ({
          ...item,
          origCreated: params.data.created,
        }))
      );
    },
  },
});

export const getGridDefinitions = (formatDateTime, context) => ({
  ...getMasterDetailsConf(formatDateTime, context),
  components,
  columnDefs: getColumnDefinitions(formatDateTime, true),
  isRowMaster: dataItem => (dataItem ? dataItem.backups : false),
});
