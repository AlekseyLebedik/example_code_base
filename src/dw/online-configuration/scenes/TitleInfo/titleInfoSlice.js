/* eslint-disable no-param-reassign */
import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import { createApiUrl } from './helpers';
import {
  getTitleInfoStats,
  getTitleInfoMMP,
  getTitleInfoEnv,
  getTitleInfoCluster,
  getTitleInfoMarketplace,
  getTitleInfoAchievements,
  getTitleInfoObjectStore,
  getMatchmaking,
  getSocialService,
} from './api';
import * as C from './constants';

const tasks = [
  { name: C.PROP_NAMES.STATS, fn: baseUrl => getTitleInfoStats(baseUrl) },
  { name: C.PROP_NAMES.MMP, fn: baseUrl => getTitleInfoMMP(baseUrl) },
  { name: C.PROP_NAMES.ENV, fn: baseUrl => getTitleInfoEnv(baseUrl) },
  {
    name: C.PROP_NAMES.MARKETPLACE,
    fn: baseUrl => getTitleInfoMarketplace(baseUrl),
  },
  {
    name: C.PROP_NAMES.ACHIEVEMENTS,
    fn: baseUrl => getTitleInfoAchievements(baseUrl),
  },
  {
    name: C.PROP_NAMES.OBJECT_STORE,
    fn: baseUrl => getTitleInfoObjectStore(baseUrl),
  },
  { name: C.PROP_NAMES.CLUSTER, fn: baseUrl => getTitleInfoCluster(baseUrl) },
  { name: C.PROP_NAMES.MATCHMAKING, fn: baseUrl => getMatchmaking(baseUrl) },
  {
    name: C.PROP_NAMES.SOCIAL_SERVICE,
    fn: baseUrl => getSocialService(baseUrl),
  },
];
export const fetchTitleInfoData = createAsyncThunk(
  'fetchTitleInfoData',
  async ({ name, fn, baseUrl }) => {
    const response = {};
    const { data } = await fn(baseUrl);
    response[name] = data;
    return response;
  }
);

export const fetchTitleInfo = createAsyncThunk(
  'fetchTitleInfo',
  (_, { dispatch, getState }) => {
    const state = getState();
    const baseUrl = createApiUrl(state);
    tasks.forEach(({ name, fn }) =>
      dispatch(fetchTitleInfoData({ name, fn, baseUrl }))
    );
  }
);

const initialData = {
  titleInfo: {},
  loading: false,
};
const titleInfoSlice = createSlice({
  name: 'titleInfo',
  initialState: initialData,
  reducers: {},
  extraReducers: {
    [fetchTitleInfoData.pending]: state => {
      state.loading = true;
    },
    [fetchTitleInfoData.fulfilled]: (state, action) => {
      state.titleInfo = { ...state.titleInfo, ...action.payload };
      state.loading = false;
    },
  },
});

// reducer
export default titleInfoSlice.reducer;
