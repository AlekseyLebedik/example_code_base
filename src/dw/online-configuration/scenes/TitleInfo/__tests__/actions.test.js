import { nonCriticalHTTPError } from '@demonware/devzone-core/helpers/errors';
import { fetchTitleInfo } from '../titleInfoSlice';

jest.mock('@demonware/devzone-core/helpers/errors');

describe('TitleInfo', () => {
  const MOCKED_NON_CRITICAL_ERROR = 'MOCKED_NON_CRITICAL_ERROR';

  describe('Actions', () => {
    beforeEach(() => {
      nonCriticalHTTPError.mockReset();
      nonCriticalHTTPError.mockReturnValue({
        type: MOCKED_NON_CRITICAL_ERROR,
      });
    });

    describe('fetchTitleInfo', () => {
      it('returns TITLE_INFO_FETCH action', () => {
        expect(fetchTitleInfo.pending()).toMatchSnapshot({
          type: fetchTitleInfo.pending.type,
        });
      });
    });

    describe('fetchTitleInfoSuccess', () => {
      it('returns TITLE_INFO_FETCH_SUCCESS action', () => {
        expect(fetchTitleInfo.fulfilled()).toMatchSnapshot({
          type: fetchTitleInfo.fulfilled.type,
        });
      });
    });
  });
});
