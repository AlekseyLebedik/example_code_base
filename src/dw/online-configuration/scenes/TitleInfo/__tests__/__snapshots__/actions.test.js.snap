// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleInfo Actions fetchTitleInfo returns TITLE_INFO_FETCH action 1`] = `
Object {
  "meta": Object {
    "arg": undefined,
    "requestId": undefined,
    "requestStatus": "pending",
  },
  "payload": undefined,
  "type": "fetchTitleInfo/pending",
}
`;

exports[`TitleInfo Actions fetchTitleInfoSuccess returns TITLE_INFO_FETCH_SUCCESS action 1`] = `
Object {
  "meta": Object {
    "arg": undefined,
    "requestId": undefined,
    "requestStatus": "fulfilled",
  },
  "payload": undefined,
  "type": "fetchTitleInfo/fulfilled",
}
`;
